+nosnoc/+solver/MpccSolver.m

```diff
@@ -286,9 +608,21 @@
             [~, min_idx] = sort(min_dists);
 
             if complementarity_constraints_lifted
-                w = nlp.w.mpcc_w(); lbw_orig = nlp.w.mpcc_w().lb; ubw_orig = nlp.w.mpcc_w().ub;
-                g = nlp.g.mpcc_g(); lbg = nlp.g.mpcc_g().lb; ubg = nlp.g.mpcc_g().ub;
-                p = nlp.p.mpcc_p();
+                w = obj.w_mpcc_fun(nlp.w.sym); lbw_orig = obj.w_mpcc_fun(nlp.w.lb); ubw_orig = obj.w_mpcc_fun(nlp.w.ub);
+                nlp_g = nlp.g.sym; g = SX(zeros([length(obj.ind_map_g.nlp), 1]));
+                nlp_lbg = nlp.g.lb; lbg = [];
+                nlp_ubg = nlp.g.ub; ubg = [];
+                if ~isempty(nlp_lbg)
+                    g(obj.ind_map_g.nlp) = nlp_g(obj.ind_map_g.mpcc)';
+                    lbg(obj.ind_map_g.nlp) = nlp_lbg(obj.ind_map_g.mpcc)';
+                    lbg = lbg';
+                    ubg(obj.ind_map_g.nlp) = nlp_ubg(obj.ind_map_g.mpcc)';
+                    ubg = ubg';
+                end
+                p = nlp.p.sym;
+                p(obj.ind_map_p.nlp) = p(obj.ind_map_p.mpcc)';
+                nlp_p_val = nlp.p.val;p_val = [];
+                p_val(obj.ind_map_p.nlp) = nlp_p_val(obj.ind_map_p.mpcc);
```

Comment:
is it really necessary to call all settings and constructions within a loop? 
lot of things do not change